ToolBox/CoinApi/CoinApiSymbolMapper.cs

```diff
@@ -0,0 +1,178 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Newtonsoft.Json;
+using QuantConnect.Brokerages;
+using QuantConnect.Configuration;
+
+namespace QuantConnect.ToolBox.CoinApi
+{
+    /// <summary>
+    /// Provides the mapping between Lean symbols and CoinAPI symbols.
+    /// </summary>
+    /// <remarks>For now we only support mapping for CoinbasePro (GDAX) and Bitfinex</remarks>
+    public class CoinApiSymbolMapper : ISymbolMapper
+    {
+        private const string RestUrl = "https://rest.coinapi.io";
+        private readonly string _apiKey = Config.Get("coinapi-api-key");
+
+        // LEAN market <-> CoinAPI exchange id maps
+        private static readonly Dictionary<string, string> MapMarketsToExchangeIds = new Dictionary<string, string>
+        {
+            { Market.GDAX, "COINBASE" },
+            { Market.Bitfinex, "BITFINEX" }
+        };
+        private static readonly Dictionary<string, string> MapExchangeIdsToMarkets =
+            MapMarketsToExchangeIds.ToDictionary(x => x.Value, x => x.Key);
+
+        private static readonly Dictionary<string, Dictionary<string, string>> CoinApiToLeanCurrencyMappings =
+            new Dictionary<string, Dictionary<string, string>>
+            {
+                {
+                    Market.Bitfinex,
```

Comment:
The dictionary is missing `Market.GDAX`, wondering if the mapping is straight, based on the `ConvertCoinApiCurrencyToLeanCurrency` implementation? Maybe we could add some unit test for `Market.GDAX` too.
Or if it is yet to be implemented, we can throw some error?

Reply:
The GDAX entry is not present in the dictionary because there is no mapping required for currencies (base and quote currency symbols for CoinAPI and GDAX are the same)